homeassistant/components/datadog/__init__.py

```diff
@@ -52,6 +53,7 @@ def setup(hass, config):
 
     initialize(statsd_host=host, statsd_port=port)
 
+    @callback
```

Comment:
This is incorrect. Only decorate as `@callback` when it is to be run inside the event loop.  This function does I/O and so should be run inside the executor. See also https://developers.home-assistant.io/docs/asyncio_categorizing_functions#the-callback-function
```suggestion
```